test/Mono.Linker.Tests.Cases/Reflection/RunClassConstructorUsedViaReflection.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Reflection;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+using System.Runtime.CompilerServices;
+
+namespace Mono.Linker.Tests.Cases.Reflection
+{
+	[SetupCSharpCompilerToUse ("csc")]
+	public class RunClassConstructorUsedViaReflection
+	{
+		public static void Main ()
+		{
+			TestRunClassConstructor ();
+			TestNonKeptStaticConstructor ();
+			TestNullWithType ();
+			TestNullWithRuntimeTypeHandle ();
+			TestDataFlowType ();
+			TestIfElseUsingRuntimeTypeHandle (1);
+			TestIfElseUsingType (1);
+		}
+
+		[Kept]
+		[RecognizedReflectionAccessPattern]
+		static void TestRunClassConstructor ()
+		{
+			RuntimeHelpers.RunClassConstructor (typeof (OnlyUsedViaReflection).TypeHandle);
+		}
+
+		[Kept]
+		static void TestNonKeptStaticConstructor ()
+		{
+			var a = new NonKeptStaticConstructorClass();
+		}
+
+		[Kept]
+		[RecognizedReflectionAccessPattern]
+		static void TestNullWithType ()
+		{
+			Type type = null;
+			RuntimeHelpers.RunClassConstructor (type.TypeHandle);
+		}
+
+		[Kept]
+		[RecognizedReflectionAccessPattern]
+		static void TestNullWithRuntimeTypeHandle ()
```

Comment:
How is this different from `TestNullWithType` - I don't mind having more test cases, but these two are nearly identical.

Reply:
Basically one of them is exercising more than one RecognizedReflectionAccessPattern but they are basically having the same behavior in code, I guess it makes more sense in the IfElse case. I will leave only those.